workflow/rules/benchmarking.smk

```diff
@@ -206,6 +206,271 @@ rule plot_strain_call_error:
+    output:
+        "results/{date}/assembly/{sample}/velvet/{sample}.contigs.fasta",
+    log:
+        "logs/{date}/velvet/{sample}.log",
+    params:
+        extra="",
+        outdir=lambda w, output: os.path.dirname(output[0]),
+    threads: 8
+    conda:
+        "../envs/velvet.yaml"
+    shell:
+        """
+        velveth {params.outdir} 21 -fastq.gz -shortPaired {input.fastq1} {input.fastq2} > {log} 2>&1
+        velvetg {params.outdir} -ins_length 150 -exp_cov 10 >> {log} 2>&1
+        mv {params.outdir}/contigs.fa {output} >> {log} 2>&1
+        """
+
+
+rule assembly_comparison_metaspades:
+    input:
+        fastq1=lambda wildcards: get_reads_after_qc(wildcards, read="1"),
+        fastq2=lambda wildcards: get_reads_after_qc(wildcards, read="2"),
+    output:
+        contigs="results/{date}/assembly/{sample}/metaspades/{sample}.contigs.fasta",
+    params:
+        outdir=lambda w, output: os.path.dirname(output[0]),
+    log:
+        "logs/{date}/metaSPAdes/{sample}.log",
+    threads: 8
+    conda:
+        "../envs/spades.yaml"
+    shell:
+        "(metaspades.py -1 {input.fastq1} -2 {input.fastq2} -o {params.outdir} -t {threads} && "
+        "mv {params.outdir}/contigs.fasta {output.contigs}) > {log} 2>&1"
+
+
+rule assembly_comparison_coronaspades:
+    input:
+        fastq1=lambda wildcards: get_reads_after_qc(wildcards, read="1"),
+        fastq2=lambda wildcards: get_reads_after_qc(wildcards, read="2"),
+    output:
+        contigs="results/{date}/assembly/{sample}/coronaspades/{sample}.contigs.fasta",
+    params:
+        outdir=lambda w, output: os.path.dirname(output[0]),
+    log:
+        "logs/{date}/coronaSPAdes/{sample}.log",
+    threads: 8
+    conda:
+        "../envs/spades.yaml"
+    shell:
+        "(coronaspades.py -1 {input.fastq1} -2 {input.fastq2} -o {params.outdir} -t {threads} && "
+        "mv {params.outdir}/contigs.fasta {output.contigs}) > {log} 2>&1"
+
+
+rule assembly_comparison_spades:
+    input:
+        fastq1=lambda wildcards: get_reads_after_qc(wildcards, read="1"),
+        fastq2=lambda wildcards: get_reads_after_qc(wildcards, read="2"),
+    output:
+        contigs="results/{date}/assembly/{sample}/spades/{sample}.contigs.fasta",
+    params:
+        outdir=lambda w, output: os.path.dirname(output[0]),
+    log:
+        "logs/{date}/SPAdes/{sample}.log",
+    threads: 8
+    conda:
+        "../envs/spades.yaml"
+    shell:
+        "(spades.py -1 {input.fastq1} -2 {input.fastq2} -o {params.outdir} -t {threads} && "
+        "mv {params.outdir}/contigs.fasta {output.contigs}) > {log} 2>&1"
+
+
+rule assembly_comparison_rnaviralspades:
+    input:
+        fastq1=lambda wildcards: get_reads_after_qc(wildcards, read="1"),
+        fastq2=lambda wildcards: get_reads_after_qc(wildcards, read="2"),
+    output:
+        contigs=(
+            "results/{date}/assembly/{sample}/rnaviralspades/{sample}.contigs.fasta"
```

Comment:
all the different spades rules look exactly the same. Just have one, and turn the type into a wildcard
```suggestion
            "results/{date}/assembly/{sample}/{spadesflavor}/{sample}.contigs.fasta"
```

Then, in the shell command start with
```
shell:
    "{wildcards.spadesflavor}.py ..."
```